sklearn/pipeline.py

```diff
@@ -1727,6 +1752,33 @@ def __getitem__(self, name):
             raise KeyError("Only string keys are supported")
         return self.named_transformers[name]
 
+    def get_metadata_routing(self):
+        """Get metadata routing of this object.
+
+        Please check :ref:`User Guide <metadata_routing>` on how the routing
+        mechanism works.
+
+        .. versionadded:: 1.5
+
+        Returns
+        -------
+        routing : MetadataRouter
+            A :class:`~sklearn.utils.metadata_routing.MetadataRouter` encapsulating
+            routing information.
+        """
+        router = MetadataRouter(owner=self.__class__.__name__)
+
+        for name, transformer in self.transformer_list:
+            router.add(
+                **{name: transformer},
+                method_mapping=MethodMapping()
+                .add(callee="fit", caller="fit")
+                .add(callee="fit_transform", caller="fit")
+                .add(callee="fit_transform", caller="fit_transform"),
```

Comment:
we don't do explicit `fit_transform` here, since it's a compound method.

Reply:
Now I see:

From FeatureUnion's `fit_transform` we need to prepare for routing into three directions: to `fit`, to `transform` and to `fit_transform`. This is because of the way `_fit_transform_one` is designed:
```python
        if hasattr(transformer, "fit_transform"):
            res = transformer.fit_transform(X, y, **params.get("fit_transform", {}))
        else:
            res = transformer.fit(X, y, **params.get("fit", {})).transform(
                X, **params.get("transform", {})
            )
```
It is designed this way, because a transformer can use either `fit` and `transform` or `fit_transform` and in fact, metadata routing is only possible, if it overwrites `TransformerMixin`'s `fit_transform`.

But yes, we don't route from the FeaturUnion's fit to the estimator's `fit_transform`.

I've changed the routing to my new understandings. We actually don't have a test yet for ConsumingClassifier.fit().transform(), but I think I should go ahead and make one.

Reply:
We actually don't have a test for transformer.fit().transform(). I made one using a new class `ConsumingNoFitTransformTransformer`. It has a long name, but an accurat one. :)

This test could not be done with `ConsumingClassifier`, because it has a `fit_transform`, that would be used instead, and it could neither be done by a class that inherits from `TransformerMixin`, because it's `fit_transform` would also come first and it doesn't route metadata `transform`.